scripts/pr.py

```diff
@@ -0,0 +1,273 @@
+        data = {'githubName': gitName()}
+        checkedDict = mapDict(validateValues, data)
+        checkedDict['remote'] = requireRemote()
+        globalConfig = { scriptPath: checkedDict }
+        
+        return globalConfig
+
+    
+    
+class Tasks:
+    def __init__(self, config):
+        self.config = config
+
+        
+    def createBranch(self):
+        branchName = "dev-{0}-{1}".format(self.config.name, requireFeatureName())
+        execCmd(["git", "checkout", "-b", branchName])
+
+        
+    def createPr(self):
+        print "Creating PR. Make sure that you push recent changes to the origin"
+
+        repoName = self.config.remote
+        username = self.config.name
+        password = requirePass()
+        github = Github(username, password)
+
+        for repo in github.get_user().get_repos():
+            if repo.full_name.lower() == repoName:
+                self.checkAndMakePr(repo)
+
+
+    def requireTitle(self):
+        tickedId = requireTickedId()
+        tickedHeader = requireTickedHeader()
+        return "JIRA# {0}: {1}".format(tickedId, tickedHeader)
+
+    
+    def getHead(self):
+        return "{0}:{1}".format(self.config.name, gitCurrentBranch())
+    
+                
+    def requirePrParams(self):
+        base = requireBase()
+        head = self.getHead()
+        title = self.requireTitle()
+        body = requireBody()
+
+        return (title, body, base, head)
+
+    
+    def checkAndMakePr(self, repo):
+        if not self.test() is 0:
+            raise ValueError("Some tests are failed.")
+
+        params = self.requirePrParams()
+
+        repo.create_pull(*params)
+        
+        print "Success"
+        
+
+    def defaultScenario(self):
+        print 'Not a supported task. Use "help" command for help.'
+
+        
+    def buildSnapshot(self):
+        "Start building snapshot.."
+        return execCmd(["sbt", "package"])
+
+    
+    def test(self):
+        print "Start testing.."
+        return execCmd(["sbt", "test"])
+
+    
+    def noArgs(self):
+        print 'No arguments specified. Use "help" command for help.'
+
+        
+    def help(self):
+        print  """Commands:
+        pr       - create a pull request to the 'origin'"
+        test     - run sbt tests
+        branch   - create a new feature branch
+        snapshot - build jar snapshot"""
+        
+    
+    tasks = {
+        'snapshot': buildSnapshot,
+        'branch': createBranch,
+        'test': test,
+        'help': help,
+        'pr': createPr
+    }
+
+
+    def executeTask(self, args):
+        if len(args) == 0:
+            self.noArgs()
+        else:
+            taskName = args[0]        
+            task = self.tasks.get(taskName, Tasks.defaultScenario)
+            task(self)
+
+            
+def main():
+    gitPython = "PyGitHub"
+    
+    config = Config()
+    print"Config is", config.config
+    tasks = Tasks(config)
+    tasks.executeTask(sys.argv[1:])
```

Comment:
I'm confused at the usage here. Can you write a README that shows how to use this?